dnsmasq_interface.c

```diff
@@ -57,6 +57,9 @@ void FTL_new_query(unsigned int flags, char *name, struct all_addr *addr, char *
 		return;
 	}
 
+	// Store plain text domain in buffer for regex validation
+	char *domainbuffer = strdup(domain);
```

Comment:
a large comment on the code: why still using C rather than C++?
For example this could be an array and I think it would be easier to program and to read

Reply:
C++ is another language. Although it is surely somehow originating from C, I don't see it as its successor any anything like that. I agree that there are many convenience improvements in C++ (such as higher-level processing of strings, vectors, ...) but with C we have more fine grained control over performance as we do everything a bit more low-level. The embedded DNS/DHCP/TFTP server `dnsmasq` is written entirely in C and I want to avoid having mixed language program for no immediate benefits (despite, admittedly, maybe a bit more readability here and there).